processing/src/main/java/org/apache/druid/segment/QueryableIndexIndexableAdapter.java

```diff
@@ -174,7 +175,9 @@ public NestedColumnMergable getNestedColumnMergeables(String columnName)
     if (columnHolder == null) {
       return null;
     }
-    if (!(columnHolder.getColumnFormat() instanceof NestedCommonFormatColumn.Format)) {
+    final ColumnFormat format = columnHolder.getColumnFormat();
+    if (!(format instanceof NestedCommonFormatColumn.Format)
+        && !(format instanceof NestedDataComplexTypeSerde.NestedColumnFormatV4)) {
```

Comment:
nit: `!(format instanceof NestedCommonFormatColumn.Format || format instanceof NestedDataComplexTypeSerde.NestedColumnFormatV4)`

is perhaps a bit more easier to understand the intentions of for this.